### DISCUSSION TITLE
Pass x to another file for obj calculation and then read results into F

### DISCUSSION THREAD

**Discussion by Atulsingh92** [anyoptimization/pymoo#253] [Q&A]
Hello Julian, 

This is a logic I am thinking of building for multiple files that I have which calculates the objective function, if the parameters could be parsed to it. Which makes me think, can I save x and use the file externally, and then reassemble F. i.e. 

```
-> Use the x available in _evaluate method in file1
        -> save as csv file (csv1) per generation with all the parameters
        -> read this csv file into another file(f2) that has all the methods for objective function calculations.
        -> save the results of objectives into another csv (csv2) file
-> Read this csv file back into _evaluate method to assmeble F, G
```

The problem I have is that as _evaluate requires F and G to be defined within _evaluate itself along with x, I used a sleep function but could not get to export x and read back F during this sleep.

Could you please highlight, how could I do so, if at all?

**Comment by blankjul** [comment]
Yes, that is possible! Everything you do in `_evaluate` is up to you and your implementation. 
`pymoo` only needs to have the `out["F"]` (and `out["G"]` for constrained problems) being set.
How you set them, for instance, by calling more efficient C code, running third-party software, launching some HPC instances is totally up to you.
